webapp/tests/karma/ts/services/header-tabs.service.spec.ts

```diff
@@ -1,50 +1,107 @@
+      ]);
+    });
   });
 
-  it('should replace icons when provided', () => {
-    const headerTabsSettings = {
-      tasks: { icon: 'fa-whatever' },
-      reports: { resource_icon: 'some-icon' },
-      analytics: { resource_icon: 'other-icon', icon: 'fa-icon' },
-      contacts: { resource_icon: 'one-icon', icon: 'not-fa-icon' },
-    };
-
-    const tabs = service.get({ header_tabs: headerTabsSettings });
-    // @ts-ignore
-    expect(tabs).to.shallowDeepEqual([
-      { name: 'messages', icon: undefined, resourceIcon: undefined, defaultIcon: 'fa-envelope' },
-      { name: 'tasks', icon: 'fa-whatever', resourceIcon: undefined, defaultIcon: 'fa-flag' },
-      { name: 'reports', icon: undefined, resourceIcon: 'some-icon', defaultIcon: 'fa-list-alt' },
-      { name: 'contacts', icon: undefined, resourceIcon: 'one-icon', defaultIcon: 'fa-user' },
-      { name: 'analytics', icon: 'fa-icon', resourceIcon: 'other-icon', defaultIcon: 'fa-bar-chart-o' },
-    ]);
+  describe('canAccessTab()', () => {
+    it('should resolve if tab is found', async () => {
+      authService.has.resolves(true);
+
+      const messagesTab = await service.canAccessTab('messages');
+      const tasksTab = await service.canAccessTab('tasks');
+      const reportsTab = await service.canAccessTab('reports');
+      const contactsTab = await service.canAccessTab('contacts');
+      const analyticsTab = await service.canAccessTab('analytics');
+
+      expect(messagesTab).to.be.true;
+      expect(authService.has.callCount).to.equal(5);
+      expect(authService.has.args[0]).to.deep.equal([['can_view_messages', 'can_view_messages_tab']]);
+      expect(tasksTab).to.be.true;
+      expect(authService.has.args[1]).to.deep.equal([['can_view_tasks', 'can_view_tasks_tab']]);
+      expect(reportsTab).to.be.true;
+      expect(authService.has.args[2]).to.deep.equal([['can_view_reports', 'can_view_reports_tab']]);
+      expect(contactsTab).to.be.true;
+      expect(authService.has.args[3]).to.deep.equal([['can_view_contacts', 'can_view_contacts_tab']]);
+      expect(analyticsTab).to.be.true;
+      expect(authService.has.args[4]).to.deep.equal([['can_view_analytics', 'can_view_analytics_tab']]);
+    });
+
+    it('should reject if tab is not found', async () => {
+      authService.has.resolves(true);
+
+      const nullTab = await service
+        .canAccessTab(null)
+        .catch(error => {
+          const expectedError = 'HeaderTabsService :: Tab \'null\' not found, cannot determine tab permissions.';
+          expect(error.message).to.equal(expectedError);
+        });
+
+      const unknownTab = await service
+        .canAccessTab('abc')
+        .catch(error => {
+          const expectedError = 'HeaderTabsService :: Tab \'abc\' not found, cannot determine tab permissions.';
+          expect(error.message).to.equal(expectedError);
+        });
+
+      expect(nullTab).to.not.be.true;
+      expect(unknownTab).to.not.be.true;
```

Comment:
```suggestion
      expect(nullTab).to.be.false;
      expect(unknownTab).to.be.false;
```
Tiny change here to simplify the checks

Reply:
Added, thanks!

Reply:
I've removed the unit test and code as the function isn't needed anymore